### PR TITLE
Support Protobuf plugin by kanro

### PR DESCRIPTION
**Pull request by devkanro**
This PR make buf plugin support the [IntelliJ Protobuf Plugin](https://github.com/devkanro/intellij-protobuf-plugin) by @devkanro, hereafter referred to as the Protobuf plugin.

The buf plugin previously only supported the Protobuf plugin provided by JetBrains, hereafter referred to as the official plugin.

This Protobuf plugin is more feature rich than the official plugin, please see the [GitHub home page](https://github.com/devkanro/intellij-protobuf-plugin) for details..

I make both the deps of official plugin and protobuf plugin be optional, the buf plugin will work fine with anyone for them.

Resolve 
- https://github.com/devkanro/intellij-protobuf-plugin/discussions/214

### CODE REVIEW

**Comment by nicksnyder** [comment]
Thanks for the contribution! Relevant reviewers are on vacation so we can take a look after they are back. In the meantime, it looks like CI doesn't pass with this change.

**Comment by devkanro** [comment]
Currently, the IDEA test framework does not support configuring the missing optional plugin, so I did not update some tests in the protobuf plugin, and the current test plan remains unchanged (only testing when the official plugin is enabled).

- https://github.com/JetBrains/gradle-intellij-plugin/issues/115

**Comment by devkanro** [comment]
@nicksnyder Could you approval the workflows to run the tests.

File: src/main/kotlin/build/buf/intellij/annotator/BufAnalyzeModificationTracker.kt [hunk 1]
```diff
@@ -25,17 +26,19 @@ import com.intellij.protobuf.lang.PbLanguage
...
+import io.kanro.idea.plugin.protobuf.lang.ProtobufLanguage
 
 /**
  * Project level service to keep track of all changes to .proto files, Buf configuration files, and the plugin config.
  * This is used to avoid unnecessary calls to BufAnalyzePass (i.e. 'buf breaking' and 'buf lint').
  */
 @Service(Service.Level.PROJECT)
 class BufAnalyzeModificationTracker(private val project: Project) : ModificationTracker {
-    private val protoModificationTracker = PsiModificationTracker.getInstance(project).forLanguage(PbLanguage.INSTANCE)
+    private val protoModificationTracker = PsiModificationTracker.getInstance(project).forLanguage(protobufLanguage()!!)
```
Comment (nicksnyder) [inline src/main/kotlin/build/buf/intellij/annotator/BufAnalyzeModificationTracker.kt:37]: I am not familiar with Kotlin, but I assume this a forcefully unwrapping the optional and will hard crash if neither is present. Can you figure out how to either ensure one is always defined or gracefully handling this situation?

**Review by nicksnyder — COMMENTED** [review]
There are some check warnings on this diff. You can see them inline when you view the PR diff or you can see the check output here: https://github.com/bufbuild/intellij-buf/actions/runs/7338528883

**Comment by devkanro** [comment]
@nicksnyder Could you approval the workflows to run the tests again.

File: build.gradle.kts [hunk 2]
```diff
@@ -189,13 +191,35 @@ tasks {
         password = environment("PRIVATE_KEY_PASSWORD")
     }
 
+    withType<PrepareSandboxTask> {
+        val vendor = project.findProperty("buf.protobuf.vendor")?.toString()
+            ?: System.getenv("BUF_PROTOBUF_VENDOR") ?: "official"
+        logger.info("Current Protobuf Plugin vendor: $vendor")
+        val file = project.layout.buildDirectory.dir(configDir).get().file("disabled_plugins.txt").asFile
```
Comment (pkwarren) [inline build.gradle.kts:198]:
```suggestion
        val disabledPluginsFile = project.layout.buildDirectory.dir(configDir).get().file("disabled_plugins.txt").asFile
```

File: gradle.properties [hunk 3]
```diff
@@ -7,14 +7,14 @@ pluginRepositoryUrl=https://github.com/bufbuild/intellij-buf
 pluginVersion=0.2.1
 # See https://plugins.jetbrains.com/docs/intellij/build-number-ranges.html
 # for insight into build numbers and IntelliJ Platform versions.
-pluginSinceBuild=223
+pluginSinceBuild=233
```
Comment (pkwarren) [inline gradle.properties:10]: Do we need to bump to the latest? We had been supporting several older versions so far.

While we're in here, we should bump `pluginVersion` above to 0.3.0.

File: gradle.properties [hunk 4]
```diff
@@ -7,14 +7,14 @@ pluginRepositoryUrl=https://github.com/bufbuild/intellij-buf
...
 pluginUntilBuild=999.*
 # IntelliJ Platform Properties -> https://github.com/JetBrains/gradle-intellij-plugin#intellij-platform-properties
 platformType=IU
-platformVersion=2022.3.3
+platformVersion=2023.3
 # Plugin Dependencies -> https://plugins.jetbrains.com/docs/intellij/plugin-dependencies.html
 # Example: platformPlugins = com.intellij.java, com.jetbrains.php:203.4449.22
-platformPlugins=idea.plugin.protoeditor, org.jetbrains.plugins.yaml
+platformPlugins=idea.plugin.protoeditor, io.kanro.idea.plugin.protobuf:1.7.50, org.jetbrains.plugins.yaml
```
Comment (pkwarren) [inline gradle.properties:17]: How will this impact users installing this plugin? Will they be asked to install one or the other?

Comment (devkanro) [reply gradle.properties:17]: No, those deps used for compiling and coding.

Comment (devkanro) [reply gradle.properties:17]: Users just need install one of protobuf plugins can be work

File: build.gradle.kts [hunk 5]
```diff
@@ -189,13 +191,35 @@ tasks {
         password = environment("PRIVATE_KEY_PASSWORD")
     }
 
+    withType<PrepareSandboxTask> {
+        val vendor = project.findProperty("buf.protobuf.vendor")?.toString()
+            ?: System.getenv("BUF_PROTOBUF_VENDOR") ?: "official"
+        logger.info("Current Protobuf Plugin vendor: $vendor")
+        val disabledPluginsFile = project.layout.buildDirectory.dir(configDir).get().file("disabled_plugins.txt").asFile
```
Comment (pkwarren) [inline build.gradle.kts:198]: Sorry for the previous suggestion - you'll need to modify the `file` variable below to use the new name as well.

File: gradle.properties (hunk 3 above)
Comment (devkanro) [reply gradle.properties]: I have checked compatibility with devkanro's protobuf plugin 1.6.32(target the 2022.3)

**Comment by devkanro** [comment]
@pkwarren Plz approval the workflows to run the tests again.

**Comment by Alfus** [comment]
FYI, we filed https://github.com/devkanro/intellij-protobuf-plugin/issues/229 as a potental avenue for deeper integration.

**Review by nicksnyder — APPROVED** [review]
Thank you for contributing this!

**Pull request merged**